### PR TITLE
[DOCS] Add put snapshot repo API docs

### PR DESCRIPTION
**Pull request by jrodewig**
Relates to #56069

### CODE REVIEW

File: docs/reference/snapshot-restore/apis/put-repo-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+.Valid `settings` properties for `fs` repositories
+[%collapsible%open]
+====
+`chunk_size`::
+(Optional, <<byte-units,byte value>>)
+Maximum size of files in snapshots. In snapshots, files larger than this are
+broken down into chunks of this size or smaller. Defaults to `null` (unlimited
+file size).
+
+`compress`::
```
Comment (original-brownbear) [inline docs/reference/snapshot-restore/apis/put-repo-api.asciidoc:177]: Should we just leave this out of the documentation maybe? This is a really useless setting at this point. Having it documented has led to some confusion in the past and I can't see what a user would get out of setting this to `false`.
@DaveCTurner @tlrx any thoughts on this?

File: docs/reference/snapshot-restore/apis/put-repo-api.asciidoc [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
++
+If `false`, the cluster can write to the repository and create snapshots in it.
+Defaults to `false`.
++
+TIP: If you register the same snapshot repository with multiple clusters, only
+one cluster should have write access to the repository. Only a cluster with
+write access can create snapshots in the repository. All other clusters
+connected to the repository should have the `readonly` parameter set to `true`.
+This means those clusters can retrieve or restore snapshots from the repository
+but not create snapshots in it.
```
Comment (original-brownbear) [inline docs/reference/snapshot-restore/apis/put-repo-api.asciidoc:210]: Maybe add a sentence here, along the lines of "Having multiple clusters write to the repository at the same time risks corrupting the contents of the repository".

**Review by original-brownbear — APPROVED** [review]
LGTM thanks! Suggested one possible addition (the multiple clusters write at the same time bit) and a change we can make later that I just wanted to raise to a wider audience (removing compress documentation).

File: docs/reference/snapshot-restore/apis/put-repo-api.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/snapshot-restore/apis/put-repo-api.asciidoc]: Thanks! Addressed with 9cecaff01ce.

File: docs/reference/snapshot-restore/apis/put-repo-api.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/snapshot-restore/apis/put-repo-api.asciidoc:177]: I'll wait for more feedback, but if we remove this, I'll also remove the reference here:
https://www.elastic.co/guide/en/elasticsearch/reference/master/snapshots-register-repository.html#snapshots-filesystem-repository

**Comment by jrodewig** [comment]
Thanks @original-brownbear. I'll go ahead and merge this. If we decide to go that direction, I'll remove the `compress` param docs as part of a later effort.

**Comment by jrodewig** [comment]
### Backport commits
master 76e45d9 
7.x 74554f1
7.8 6d9c2c2 
7.7 80bcd0a

**Pull request merged**